src/test/java/org/jenkinsci/plugins/workflow/actions/ErrorActionTest.java

```diff
@@ -0,0 +1,110 @@
+import org.jenkinsci.plugins.workflow.graph.FlowGraphWalker;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.BuildWatcher;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import hudson.model.Result;
+import hudson.remoting.ProxyException;
+
+/**
+ * Tests for {@link ErrorAction}
+ */
+public class ErrorActionTest {
+    @ClassRule
+    public static BuildWatcher buildWatcher = new BuildWatcher();
+
+    @Rule
+    public JenkinsRule r = new JenkinsRule();
+
+    private List<ErrorAction> extractErrorActions(FlowExecution exec) {
+        List<ErrorAction> ret = new ArrayList<ErrorAction>();
+
+        FlowGraphWalker walker = new FlowGraphWalker(exec);
+        for (FlowNode n : walker) {
+            ErrorAction e = n.getAction(ErrorAction.class);
+            if (e != null) {
+                ret.add(e);
+            }
+        }
+        return ret;
+    }
+
+    @Test
+    public void simpleException() throws Exception {
+        final String EXPECTED = "For testing purpose";
+        WorkflowJob job = r.jenkins.createProject(WorkflowJob.class, r.getTestDescription().getMethodName());
+        job.setDefinition(new CpsFlowDefinition(String.format(
+                "node {\n"
+                        + "throw new Exception('%s');\n"
+                + "}"
+                , EXPECTED
+        )));
+        WorkflowRun b = r.assertBuildStatus(Result.FAILURE, job.scheduleBuild2(0).get());
+        List<ErrorAction> errorActionList = extractErrorActions(b.asFlowExecutionOwner().get());
+        assertThat(errorActionList, Matchers.not(Matchers.empty()));
+        for (ErrorAction e : errorActionList) {
+            assertEquals(Exception.class, e.getError().getClass());
+            assertEquals(EXPECTED, e.getError().getMessage());
+        }
+    }
+
+    @Issue("JENKINS-34488")
+    @Test
+    public void unserializableForSecurityReason() throws Exception {
+        WorkflowJob job = r.jenkins.createProject(WorkflowJob.class, r.getTestDescription().getMethodName());
+        // "assert false" throws org.codehaus.groovy.runtime.powerassert.PowerAssertionError,
+        // which is rejected by remoting.
+        job.setDefinition(new CpsFlowDefinition(
+                "node {\n"
+                        + "assert false;\n"
+                + "}"
+        ));
+        WorkflowRun b = r.assertBuildStatus(Result.FAILURE, job.scheduleBuild2(0).get());
+        List<ErrorAction> errorActionList = extractErrorActions(b.asFlowExecutionOwner().get());
+        assertThat(errorActionList, Matchers.not(Matchers.empty()));
+        for (ErrorAction e : errorActionList) {
+            assertEquals(ProxyException.class, e.getError().getClass());
```

Comment:
More interesting would be to have a script which runs, say

```groovy
assert 2 + 2 == 5
```

and verify that the build log contains (`r.assertLogContains`) the expected output from `PowerAssertionError` (at least the number `4` somewhere).